Sources/Boundaries/Effect.swift

```diff
@@ -4,12 +4,12 @@ public enum Effect<A> {
   case batch([Effect<A>])
   case dispatch(A)
   case sequence([Effect<A>])
-  case _execute(String, Any, (Any) -> A?)
+  case _execute(String, Any, (Any, @escaping (A) -> ()) -> ())
```

Comment:
Alternatively, `_execute` could be:

``` swift
case _execute(/* _ tag */: [Any], (@escaping (A) -> ()) -> ())
```

Though this puts it (slightly) more on the effect creator to make sure they pass all relevant data to `tag`.